Formatter/Pool.php

```diff
@@ -34,15 +34,13 @@ class Pool
     protected $logger;
 
     /**
-     * @param LoggerInterface|null $logger
      * @param string|null          $defaultFormatter
+     * @param LoggerInterface|null $logger
      */
-    public function __construct(LoggerInterface $logger = null, $defaultFormatter = null)
+    public function __construct($defaultFormatter, LoggerInterface $logger = null)
```

Comment:
Null is valid on the 3.x branch IMO, since it's optional. When it is null, the first one is automatically selected, this is necessary for BC. Otherwise I agree, deprecation warnings should be added. (Actually I am not sure why I didn't add any)